File: landing/src/main/java/com/walhalla/landing/ChromeView.java
package com.walhalla.landing;

import android.webkit.WebView;

public interface ChromeView {
    void onPageStarted(String url);

    void onPageFinished(/*WebView view, */String url);

    void webClientError(int errorCode, String description, String failingUrl);

    void removeErrorPage();

    void setErrorPage();

    //void mAcceptPressed(String url);

    //void eventRequest(BodyClass bodyClass);
}

File: features/webview/src/main/java/com/walhalla/webview/ChromeView.kt
package com.walhalla.webview

import android.app.Activity

interface ChromeView {
    fun onPageStarted(url: String?)

    fun onPageFinished( /*WebView view, */url: String?)

    fun webClientError(failure: ReceivedError?)

    fun removeErrorPage()

    fun setErrorPage(receivedError: ReceivedError?)

    fun openBrowser(
        context: Activity?,
        url: String?
    )
    //void openOauth2(Activity context, String url);
    //void mAcceptPressed(String url);
    //void eventRequest(BodyClass bodyClass);
}